libs/sdk-bindings/bindings-android/lib/src/main/kotlin/breez_sdk_notification/job/RedeemSwap.kt

```diff
@@ -33,32 +34,43 @@ class RedeemSwapJob(
     override fun start(breezSDK: BlockingBreezServices) {
         try {
             val request = Json.decodeFromString(AddressTxsConfirmedRequest.serializer(), payload)
+            this.bitcoinAddress = request.address
             breezSDK.redeemSwap(request.address)
             logger.log(TAG, "Found swap for ${request.address}", "INFO")
-            notifyChannel(
-                context,
-                NOTIFICATION_CHANNEL_SWAP_TX_CONFIRMED,
-                getString(
-                    context,
-                    SWAP_TX_CONFIRMED_NOTIFICATION_TITLE,
-                    DEFAULT_SWAP_TX_CONFIRMED_NOTIFICATION_TITLE
-                ),
-            )
         } catch (e: Exception) {
-            logger.log(TAG, "Failed to process swap notification: ${e.message}", "WARN")
-            notifyChannel(
-                context,
-                NOTIFICATION_CHANNEL_SWAP_TX_CONFIRMED,
-                getString(
-                    context,
-                    SWAP_TX_CONFIRMED_NOTIFICATION_FAILURE_TITLE,
-                    DEFAULT_SWAP_TX_CONFIRMED_NOTIFICATION_FAILURE_TITLE
-                ),
-            )
+            logger.log(TAG, "Failed to manually reedeem swap notification: ${e.message}", "WARN")
         }
+    }
 
-        fgService.shutdown()
+    override fun onEvent(e: BreezEvent) {
+        this.bitcoinAddress?.let {address ->
+            when (e) {
+                is BreezEvent.SwapUpdated -> {
+                    val swapInfo = e.details
+                    logger.log(TAG, "Received swap updated event: ${swapInfo.bitcoinAddress} current address: ${address} status: ${swapInfo.status}", "TRACE")
+                    if (swapInfo.bitcoinAddress == address) {
+                        if (swapInfo.paidMsat.toLong() > 0) {
+                            notifySuccessAndShutdown()
+                        }
+                    }
+                }
+
+                else -> {}
+            }
+        }
     }
 
-    override fun onEvent(e: BreezEvent) {}
+    private  fun notifySuccessAndShutdown() {
+        logger.log(TAG, "Swap address redeemed succesfully", "INFO")
```

Comment:
We do log address information on SwapUpdated event but I suggest we log which address was successful here as well.
```suggestion
    private  fun notifySuccessAndShutdown(address: String) {
        logger.log(TAG, "Swap address ${address} redeemed successfully", "INFO")
```